[Jeron Kor] iP

jeronkk opened the pull request:
# **LeChatBot 🤖**  

> *"[You can't be afraid to fail. It's the only way you succeed - you're not gonna succeed all the time, and I know that.]"*  

LeChatBot is a **lightweight, text-based chatbot** that helps you manage your tasks, deadlines, and events seamlessly.  
It's:  
✔ **Text-based** for simplicity  
✔ **User-friendly** and **intuitive**  
✔ **Fast** and **efficient**  

## **🚀 Getting Started**  
To start using LeChatBot, simply:  
1️⃣ Download it from **[here](#)**.  
2️⃣ Double-click to launch it.  
3️⃣ Add your tasks, deadlines, and events.  
4️⃣ Let **LeChatBot** handle the rest! 😃  

And best of all, it's **FREE!** 🎉  

## **✨ Features**  
✅ Managing tasks efficiently  
✅ Adding deadlines and event scheduling  
✅ Searching for tasks with keywords  
✅ Marking and unmarking tasks as completed  
✅ Deleting tasks with ease  
✅ Friendly UI with clear prompts  

### 🔜 Upcoming Features  
🔲 Recurring reminders (coming soon)  
🔲 Task prioritization (coming soon)  

---

## **📌 Example Usage**  
```sh
todo Trade Anthony Davis
deadline Trade DLo /by 2024-12-28
event Team meeting /from 14:00 /to 16:00
find Bronny

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jeronkk committed:
Rename bot to LebronJames.java

jeronkk committed:
Add LebronJames ascii art and text

jeronkk committed:
Add greeting and exiting messages

jeronkk committed:
Add scanner to implement echo

jeronkk committed:
Implement task listing functionality with handling for empty task list

jeronkk committed:
Add task addition functionality with a limit of 100 tasks

jeronkk committed:
Add Task class with status tracking and string representation of task

jeronkk committed:
Refactor task storage to use Task objects and improve UI formatting

jeronkk committed:
Add mark and unmark commands for task completion tracking

jeronkk committed:
Refactor task operations into separate methods for better modularity

jeronkk committed:
Refactor Task class for encapsulation and add Deadline, Event, and Todo subclasses

jeronkk committed:
Add task handlers for Todo, Deadline, and Event types

jeronkk committed:
Update test script to use LebronJames instead of Duke

jeronkk committed:
Fix line-ending issues in test script by converting output to UNIX format

jeronkk committed:
Update expected output and add basic inputs for automated testing

jeronkk committed:
Fix formatting issue in welcome and exit messages

jeronkk committed:
Rename project references from Duke to LebronJames

jeronkk committed:
Add LebronJamesException for handling task-related errors

jeronkk committed:
Integrate LebronJamesException for better error handling in task operations

jeronkk committed:
Refactor task storage from array to ArrayList for dynamic management

jeronkk committed:
Add delete task functionality with error handling

jeronkk committed:
Update input test file to include delete task scenarios

jeronkk committed:
Update expected output to reflect delete task functionality and error handling

jeronkk committed:
Update test script to compare ACTUAL.TXT directly with EXPECTED.TXT

jeronkk committed:
Rename LebronJames to LeChatBot

jeronkk committed:
Update README to reflect LeChatBot project name

jeronkk committed:
Refactor task storage to use List and introduce enums for commands and errors

jeronkk committed:
Update test input file with additional tasks and modifications

jeronkk committed:
Add Storage class for saving and loading tasks from LeChatBot.txt

jeronkk committed:
Make Task an abstract class and add file formatting method

jeronkk committed:
Integrate storage system for task persistence in LeChatBot

jeronkk committed:
Update test script to handle persistent task storage

jeronkk committed:
Merge branch 'branch-Level-7'

jeronkk committed:
Enhance Deadline task with date-time parsing and formatting

jeronkk committed:
Update input.txt with standardized date-time formats

jeronkk committed:
Merge branch 'branch-Level-8'

jeronkk committed:
Refactor project to improve OOP structure and implement UI component

jeronkk committed:
Introduce TaskList class to manage tasks more effectively

jeronkk committed:
Add Parser class to handle user input processing

jeronkk committed:
Add abstract Command class to define structure for command execution

jeronkk committed:
Implement AddCommand and DeleteCommand for task management

jeronkk committed:
Implement MarkCommand and UnmarkCommand for task management

jeronkk committed:
Implement DeadlineCommand and EventCommand for task management

jeronkk committed:
Implement ExitCommand to handle program termination

jeronkk committed:
Implement ListCommand to display all tasks

jeronkk committed:
Refactor Storage to use an instance variable for file path

jeronkk committed:
Refactor LeChatBot to use new OOP-based command and storage structure

jeronkk committed:
Merge tag 'A-MoreOOP'

jeronkk committed:
Refactor package structure for better organization

jeronkk committed:
Update test script to reflect new package structure

jeronkk committed:
Merge branch 'add-gradle-support'

jeronkk committed:
Add JUnit tests for Parser, Storage, and TaskList classes

jeronkk committed:
Update Gradle build configuration to reflect new package structure

jeronkk committed:
Merge tag 'A-JUnit'

jeronkk committed:
Add Javadoc comments to all public and private classes for better documentation

malihahaque commented on src/main/java/lechatbot/LeChatBot.java:
```diff
@@ -0,0 +1,58 @@
+package lechatbot;
```
The code is easy to follow and uses good OOP principles to make it clear! Keep it up!

xuxuezhou commented on src/main/java/lechatbot/Ui.java:
```diff
@@ -0,0 +1,101 @@
+                ╚══════╝╚══════╝╚═════╝ ╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═══╝     ╚════╝ ╚═╝  ╚═╝╚═╝     ╚═╝╚══════╝╚══════╝
+                """;
+        System.out.println(asciiArt + asciiText);
+        showLine();
```
May consider optimizing showLine() by appending \n to the LINE constant to reduce redundant System.out.println() calls.

malihahaque commented on src/main/java/lechatbot/Parser.java:
```diff
@@ -0,0 +1,75 @@
+package lechatbot;
```
The code follows good coding standards with consistent naming conventions! Might want to add JavaDocs for public methods

xuxuezhou reviewed (commented):
Overall good coding standard and code quality, great job!

malihahaque reviewed (commented):
Looks really good and very easy to understand!

Charlesl12 commented on src/main/java/lechatbot/LeChatBot.java:
```diff
@@ -0,0 +1,58 @@
+            return message;
+        }
+    }
+
```
Great use of enum!

Charlesl12 reviewed (commented):
I like the code quality and it seems to be compliant with the coding standard! JavaDocs could make improve its readability.

jeronkk committed:
Add missing javadoc comments in LeChatBot.java

jeronkk committed:
Add missing javadoc comments for Parser.java

jeronkk committed:
Add FindCommand to support searching for tasks by keyword

jeronkk committed:
Add UI and Parser support for displaying matching tasks in search results

jeronkk committed:
Merge branch 'branch-A-JavaDoc'

jeronkk committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/lechatbot/LeChatBot.java
#	src/main/java/lechatbot/Parser.java

jeronkk committed:
Merge branch 'branch-Level-9'

jeronkk committed:
Add JavaFX dependencies to build.gradle for GUI support

jeronkk committed:
Implement GUI components using JavaFX

jeronkk committed:
Add CSS styling for JavaFX UI components

jeronkk committed:
Add UI resources: FXML and CSS files for JavaFX

jeronkk committed:
Add image assets for UI: Bronny and LeBron

jeronkk committed:
Refactor LeChatBot: Add command type tracking and response generation

jeronkk committed:
Refactor Command class to return execution response

jeronkk committed:
Update expected output after OOP refactor

jeronkk committed:
Merge branch 'branch-Level-10'

jeronkk committed:
Add Checkstyle configuration files

jeronkk committed:
Integrate Checkstyle into Gradle build

jeronkk committed:
Apply Checkstyle formatting to all Java files

jeronkk committed:
Merge branch 'branch-A-CheckStyle'

jeronkk committed:
Integrate JavaFX and Refactor UI Components

Current Situation
The UI components are not well-structured, with style logic embedded within Java classes. The main application class is still LeChatBot, and JavaFX dependencies are not explicitly declared in build.gradle. Additionally, Checkstyle violations are present across multiple files.

Why It Needs to Change
The current structure makes the UI harder to maintain and extend. Separating styling concerns improves readability and maintainability. Declaring JavaFX dependencies explicitly ensures compatibility across different environments. Fixing Checkstyle violations improves code quality and adherence to coding standards.

What Is Being Done About It
* Add org.openjfx.javafxplugin to build.gradle and configure JavaFX modules.
* Change application.mainClass from lechatbot.LeChatBot to lechatbot.ui.Main.
* Set minimum stage dimensions in Main.java to enhance UI presentation.
* Remove changeDialogStyle() from DialogBox.java and rely on CSS for styling.
* Update MainWindow.java to display an introductory message upon startup.
* Rename setDuke() to setLeChatBot() for consistency.
* Modify handleUserInput() to streamline response handling.
* Fix Checkstyle violations and improve JavaDocs across modified files.

Why It Is Done That Way
* Moving styling logic to CSS ensures separation of concerns and a cleaner codebase.
* Using lechatbot.ui.Main as the main entry point better aligns with JavaFX architecture.
* Setting stage dimensions improves the default UI experience.
* Fixing Checkstyle violations enforces consistent code formatting and readability.

Other Relevant Info
This refactor improves maintainability and scalability while ensuring JavaFX compatibility. Future enhancements can build upon this cleaner and more structured UI foundation.

jeronkk committed:
Clear saved tasks after test execution

jeronkk committed:
Add assertions in Parser.java to validate task details

jeronkk committed:
Merge pull request #3 from jeronkk/branch-A-CodeQuality

Improve Code Quality and refactor UI Components

jeronkk committed:
Refactor to remove all traces of Duke

jeronkk committed:
Merge remote-tracking branch 'origin/master' into branch-A-Assertions

# Conflicts:
#	src/main/java/lechatbot/ui/Main.java
#	src/main/java/lechatbot/ui/MainWindow.java

jeronkk committed:
Merge pull request #2 from jeronkk/branch-A-Assertions

Add assertions in Parser.java to validate task details

jeronkk committed:
Add HelpCommand to display available commands

jeronkk committed:
Add support for help command in Parser

jeronkk committed:
Merge branch 'BCD-Extension'

jeronkk committed:
Improve help message formatting and content

jeronkk committed:
Add UI screenshot for documentation

jeronkk committed:
Update README with product details and usage examples

jeronkk committed:
Merge tag 'A-UserGuide'

jeronkk committed:
Update README.md to include Ui.png

jeronkk committed:
Remove wrong UI.png

jeronkk committed:
Add new Ui.png

jeronkk committed:
Add Launcher.java to launch application

jeronkk committed:
Refactor build configuration and update main entry point
